src/hb-open-file-private.hh

```diff
@@ -286,6 +286,205 @@ struct TTCHeader
+  {
+    TRACE_SANITIZE (this);
+    // actual data sanitization is done on ResourceForkHeader sanitizer
+    return_trace (c->check_struct (this));
+  }
+
+  inline unsigned int get_data_offset () const
+  {
+    return dataOffset;
+  }
+
+  inline int get_name_offset () const
+  {
+    return nameOffset;
+  }
+
+  protected:
+  HBINT16	id;		/* Resource ID, is really should be signed? */
+  HBINT16	nameOffset;	/* Offset from beginning of resource name list
+				 * to resource name, minus means there is no */
+  HBUINT8	attr;		/* Resource attributes */
+  HBUINT24	dataOffset;	/* Offset from beginning of resource data to
+				 * data for this resource */
+  HBUINT32	reserved;	/* Reserved for handle to resource */
+  public:
+  DEFINE_SIZE_STATIC (12);
+};
+
+struct ResourceTypeItem
+{
+  inline bool sanitize (hb_sanitize_context_t *c) const
+  {
+    TRACE_SANITIZE (this);
+    // RefList sanitization is done on ResourceMap sanitizer
+    return_trace (c->check_struct (this));
+  }
+
+  inline unsigned int get_number_of_resource () const
+  {
+    return numRes + 1;
+  }
+
+  inline bool is_sfnt () const
+  {
+    return type[0] == 's' && type[1] == 'f' && type[2] == 'n' && type[3] == 't';
+  }
+
+  inline const ResourceRefItem& get_ref_item (const void *base,
+					      unsigned int i) const
+  {
+    return (base+refList)[i];
+  }
+
+  protected:
+  HBUINT8	type[4];	/* Resource type */
+  HBUINT16	numRes;		/* Number of resource this type in map minus 1 */
+  OffsetTo<UnsizedArrayOf<ResourceRefItem> >
+		refList;	/* Offset from beginning of resource type list
+				 * to reference list for this type */
+  public:
+  DEFINE_SIZE_STATIC (8);
+};
+
+struct ResourceMap
+{
+  inline bool sanitize (hb_sanitize_context_t *c) const
+  {
+    TRACE_SANITIZE (this);
+    if (!c->check_struct (this))
+      return_trace (false);
+    for (unsigned int i = 0; i < get_number_of_types (); ++i)
+    {
+      const ResourceTypeItem& type = get_type (i);
+      if (!type.sanitize (c))
+        return_trace (false);
+      for (unsigned int j = 0; j < type.get_number_of_resource (); ++j)
+	if (!get_ref_item (type, j).sanitize (c))
+	  return_trace (false);
+    }
+    return_trace (true);
+  }
+
+  inline const ResourceTypeItem& get_type (unsigned int i) const
+  {
+    // Why offset from the second byte of the object? I'm not sure
+    return ((&reserved[2])+typeList)[i];
+  }
+
+  inline unsigned int get_number_of_types () const
+  {
+    return nTypes + 1;
+  }
+
+  inline const ResourceRefItem &get_ref_item (const ResourceTypeItem &type,
+					      unsigned int i) const
+  {
+    return type.get_ref_item (&(this+typeList), i);
+  }
+
+  inline const ArrayOf<HBUINT8, HBUINT8>& get_name (const ResourceRefItem &item,
+						    unsigned int i) const
+  {
+    if (item.get_name_offset () == -1)
+      return Null<ArrayOf<HBUINT8, HBUINT8> >();
+
+    return (ArrayOf<HBUINT8, HBUINT8>&) ((this+nameList)[item.get_name_offset ()]);
+  }
```

Comment:
`get_name` is not used now or ever but just to show the implementation is complete.